api/pkg/client/mlflow/mlflow.go

```diff
@@ -0,0 +1,188 @@
+package mlflow
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type mlflowClient struct {
+	Client *http.Client
+	Config Config
+}
+
+type Config struct {
+	TrackingURL string
+}
+
+type Mlflow interface {
```

Comment:
Perhaps the type definitions can go into a separate file `types.go` for better logical separation. So this file can just have the `Mlflow` interface and its implementation:

```golang
type Mlflow interface {
...
}

type mlflowClient struct {
	Client *http.Client
	Config Config
}

func NewMlflowClient(httpClient *http.Client, config Config) *mlflowClient {
	return &mlflowClient{
		Client: httpClient,
		Config: config,
	}
}

func (mfc *mlflowClient) httpCall(method string, url string, headers map[string]string, body []byte, response interface{}) error {
...
```